src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,129 @@
+
+        if (this.sqlQuery) {
+            const blockHashObservables$: Observable<string>[] = [];
+            this.queryExplainerService
+                .proccessQuery(this.sqlQuery)
+                .pipe(
+                    tap((response) => {
+                        this.sqlQueryExplainer = response;
+                        response.input.datasets
+                            .map((dataset) => ({ datasetId: dataset.id, blockHash: dataset.blockHash }))
+                            .forEach(({ datasetId, blockHash }) => {
+                                blockHashObservables$.push(
+                                    this.blockService.requestSystemTimeBlockByHash(datasetId, blockHash),
+                                );
+                            });
+                    }),
+                    switchMap((response: QueryExplainerResponse) => {
+                        return this.queryExplainerService.verifyQuery(response);
+                    }),
+                    catchError((e: HttpErrorResponse) => {
+                        return of(e.error as VerifyQueryResponse);
+                    }),
+                    tap((data) => {
+                        this.verifyResponse = data;
+                    }),
+                    switchMap(() => forkJoin(blockHashObservables$)),
+                    tap((blockHashSystemTimes) => {
+                        this.blockHashSystemTimes = blockHashSystemTimes;
+                    }),
+                    takeUntilDestroyed(this.destroyRef),
+                )
+                .subscribe(() => this.cdr.detectChanges());
+        }
+    }
+
+    public copyToClipboard(event: MouseEvent, text: string): void {
+        this.clipboard.copy(text);
+        if (event.currentTarget !== null) {
+            const currentElement: HTMLButtonElement = event.currentTarget as HTMLButtonElement;
+            const currentElementChildren: HTMLCollectionOf<HTMLElement> =
+                currentElement.children as HTMLCollectionOf<HTMLElement>;
+            setTimeout(() => {
+                currentElementChildren[0].style.display = "inline-block";
+                currentElementChildren[1].style.display = "none";
+                currentElement.classList.remove("clipboard-btn--success");
+            }, AppValues.LONG_DELAY_MS);
+            currentElementChildren[0].style.display = "none";
+            currentElementChildren[1].style.display = "inline-block";
+            currentElement.classList.add("clipboard-btn--success");
+        }
+    }
+
+    private extractSqlQueryFromRoute(): MaybeNull<string> {
+        const sqlQueryFromRoute: MaybeUndefined<string> = this.activatedRoute.snapshot.queryParams[
+            ProjectLinks.URL_QUERY_PARAM_SQL_QUERY
+        ] as MaybeUndefined<string>;
+        return sqlQueryFromRoute ?? "";
+    }
+
+    public routeToDataset(alias: string): string {
+        return alias.includes("/") ? alias : `kamu/${alias}`;
```

Comment:
Don't understand this. We have other users than `kamu`, what is the hack doing?

Reply:
If we use single-tenant image, we don't receive an account in the alias in the response.

Reply:
Consider discussing with SM a possibility of extending this output view with owner account name:

```
"datasets": [
      {
        "id": "did:odf:fed01b8083fa36c3361cc6bee416e89b09b557ff21f5fa81422295c777fbaaabd7529",
        "alias": "account.tokens.portfolio",
        "blockHash": "f1620daed817dc5d5f889af6a1dbc059e9c1a95ede3cdc11a6cee27ecd44607b0cc9b"
      }
    ],
```

Reply:
Discussed and closed